client/app/bundles/course/assessment/submission/containers/ProgrammingImportEditor.jsx

```diff
@@ -138,11 +162,16 @@ VisibleProgrammingImportEditor.propTypes = {
 };
 
 function mapStateToProps(state, ownProps) {
-  const { questionId, answerId, question, readOnly } = ownProps;
-  const { submission, form, dispatch, submissionFlags } = state;
+  const { questionId, answerId, question, readOnly, viewHistory } = ownProps;
+  const { submission, form, dispatch, submissionFlags, history } = state;
 
   const submissionId = submission.id;
-  const answers = form[formNames.SUBMISSION].values;
+  let answers;
+  if (viewHistory) {
```

Comment:
Would it be cleaner to split this into 2 containers? Most of the code is not actually shared between the two cases of `viewHistory` = true/false

Reply:
`ProgrammingImportEditor` mainly just abstracts out how it is displayed (which editor is used):
1. normal mode - can edit
2. readonly - viewing at submission
3. viewHistory - viewing past answers

While might be cleaner in that certain parts of the code, i thought it would be weird to split 1 & 2 in one container and put 3 in another when the props are almost essentially all the same, and there isn't all too much code in the file. Thoughts?

Reply:
Okay, let's leave it as it is then.